rcl/include/rcl/security.h

```diff
@@ -0,0 +1,139 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL__SECURITY_H_
+#define RCL__SECURITY_H_
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include <stdbool.h>
+
+#include "rcl/allocator.h"
+#include "rcl/types.h"
+#include "rcl/visibility_control.h"
+#include "rmw/security_options.h"
+
+#ifndef ROS_SECURITY_NODE_DIRECTORY_VAR_NAME
+  #define ROS_SECURITY_NODE_DIRECTORY_VAR_NAME "ROS_SECURITY_NODE_DIRECTORY"
+#endif
+
+#ifndef ROS_SECURITY_ROOT_DIRECTORY_VAR_NAME
+  #define ROS_SECURITY_ROOT_DIRECTORY_VAR_NAME "ROS_SECURITY_ROOT_DIRECTORY"
+#endif
+
+#ifndef ROS_SECURITY_LOOKUP_TYPE_VAR_NAME
+  #define ROS_SECURITY_LOOKUP_TYPE_VAR_NAME "ROS_SECURITY_LOOKUP_TYPE"
+#endif
+
+#ifndef ROS_SECURITY_STRATEGY_VAR_NAME
+#define ROS_SECURITY_STRATEGY_VAR_NAME "ROS_SECURITY_STRATEGY"
+#endif
+
+#ifndef ROS_SECURITY_ENABLE_VAR_NAME
+#define ROS_SECURITY_ENABLE_VAR_NAME "ROS_SECURITY_ENABLE"
+#endif
+
+/// Init the security options from the values of the environment variables and passed names.
+/**
+ * Inits the passed security options based on the environment.
+ * For more details:
+ *  \sa rcl_security_enabled
+ *  \sa rcl_get_enforcement_policy
+ *  \sa rcl_get_secure_root
+ *
+ * \param name[in] name used to find the securiy root path.
+ * \param namespace_[in] namespace_ used to find the security root path.
+ * \param allocator[in] used to do allocations.
+ * \param security_options[out] security options that will be configured according to
+ *  the environment.
+ */
+RCL_PUBLIC
+rcl_ret_t
+rcl_get_security_options_from_environment(
+  const char * name,
+  const char * namespace_,
+  const rcutils_allocator_t * allocator,
+  rmw_security_options_t * security_options);
+
+/// Check if security has to be used, according to the environment.
+/**
+ * If `ROS_SECURITY_ENABLE` environment variable is set to "true", `use_security` will be set to
+ * true.
+ *
+ * \param use_security[out] Must not be NULL.
+ * \returns RCL_RET_INVALID_ARGUMENT if an argument is not valid, or
+ * \returns RCL_RET_ERROR if an unexpected error happened, or
+ * \returns RCL_RET_OK.
+ */
+RCL_PUBLIC
+rcl_ret_t
+rcl_security_enabled(bool * use_security);
```

Comment:
Why not just return a `bool`?

Reply:
To differentiate an error condition from security being enbled or not.

Reply:
:ok_hand: